llvm/lib/Support/CommandLine.cpp

```diff
@@ -2353,9 +2353,31 @@ sortSubCommands(const SmallPtrSetImpl<SubCommand *> &SubMap,
 
 namespace {
 
+#ifdef SYCLomatic_CUSTOMIZATION
+/// HelpCategory defines various category groups for dpct options
+enum class HelpCategory {
+  HC_All,
+  HC_Basic,
+  HC_Advanced,
+  HC_CodeGen,
+  HC_ReportGen,
+  HC_BuildScript,
+  HC_QueryAPI,
+  HC_Warnings,
+  HC_HelpInfo,
+  HC_InterceptBuild,
+  HC_Examples
+};
+#endif // SYCLomatic_CUSTOMIZATION
+
+// clang-format off
+#include "llvm/migration_cmd_examples.inc"
+// clang-format on
+
 class HelpPrinter {
 protected:
   const bool ShowHidden;
+  HelpCategory helpCatEnum = HelpCategory::HC_All;
```

Comment:
for all SYCLomaitc specific code should be protected by "SYCLomatic_CUSTOMIZATION"